In [5]:
#pip install statsmodels

In [7]:
# import numpy
import numpy as np
import pandas as pd
import statsmodels.api as sm
# import linear_model and datasets from sklearn
from sklearn import linear_model, datasets

Build a regression model.

The following model hopes to make sense of 2 data points. The aim is to uncover a correlation between a dependant and independant variable.

In [8]:
#Grab our data in order to figure out what are variables are:
csv_file_path_combine = '..\data\combined_apis.csv'
csv_file_path_bikes = '..\data\df_bikes.csv'
csv_file_path_yelp = '..\data\df_yelp.csv'
csv_file_path_fsq = '..\data\df_foursquare.csv'

# Read the CSV file into a DataFrame
api_data_all = pd.read_csv(csv_file_path_combine)
df_bikes = pd.read_csv(csv_file_path_bikes)
df_yelp = pd.read_csv(csv_file_path_yelp)
df_foursquare = pd.read_csv(csv_file_path_fsq)

In [16]:
#df_bikes.tail()
df_yelp.head()
#df_foursquare.head()

city state                         name  review_count  rating  \
0  Oakland    CA               Wawa Thai Food           243     4.5   
1  Oakland    CA                      Bombera           278     4.0   
2  Oakland    CA      The Saint Sandwich Shop           239     4.5   
3  Oakland    CA                    Cafe Noir            11     5.0   
4  Oakland    CA  Phnom Penh House Restaurant           506     4.5   

    latitude   longitude    category  
0  37.797169 -122.204842        thai  
1  37.799530 -122.215200     mexican  
2  37.797250 -122.204370  sandwiches  
3  37.798756 -122.208274     burgers  
4  37.791010 -122.197760   cambodian

In [17]:
X = sm.add_constant(df_yelp[['review_count']])
y = df_yelp['rating']

model = sm.OLS(y, X).fit()

In [18]:
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                 rating   R-squared:                       0.013
Model:                            OLS   Adj. R-squared:                 -0.042
Method:                 Least Squares   F-statistic:                    0.2325
Date:                Mon, 11 Dec 2023   Prob (F-statistic):              0.636
Time:                        22:44:45   Log-Likelihood:                -27.578
No. Observations:                  20   AIC:                             59.16
Df Residuals:                      18   BIC:                             61.15
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            3.5683      0.355     10.055   

Provide model output and an interpretation of the results. 

### Interpetation

In this output, we can see that the value is 0.013 This means that the model is capable of explaining 1.3% of the patterns in the data. It may not be a great fit.


A p-value of less than 0.05 is considered to be statistically significant. This regression output shows a p-value of 0.636. In other words, review count does not impact the rating of the restaurant.

coef: In this case, we can see that the coef of review_count has a value of 0.0006. This means that a unit increase in the value of review_count will not have a positive impact on the rating of that place.

# Stretch

How can you turn the regression model into a classification model?

From my research, this would involve chunking the numerical data into bins which serve as categories to further create a new model. <br> In the case of ratings: <br> 0.0 - 2 = Poor <br> 2.1 - 3 = Average <br> 3.1 - 4 = Great <br> 4.1 - 5.0 = Exceptional

The categories that map to numbers above become the categories in which we classify our model. At least, that's my best guess.